core/arch/arm/plat-imx/drivers/tzc380.c

```diff
@@ -2,36 +2,51 @@
 /*
  * Copyright 2019 Pengutronix
  * All rights reserved.
+ * Copyright 2019 NXP
  *
  * Rouven Czerwinski <entwicklung@pengutronix.de>
  */
 
 #include <drivers/tzc380.h>
 #include <imx-regs.h>
 #include <imx.h>
+#include <kernel/panic.h>
 #include <mm/core_memprot.h>
 #include <mm/generic_ram_layout.h>
 
 void imx_configure_tzasc(void)
 {
-
-	int i;
-	vaddr_t addr[2] = {0};
+	uint8_t region;
+	vaddr_t addr[2] = { 0 };
+	uint8_t tzc_controllers = 1;
+	uint8_t i;
 
 	addr[0] = core_mmu_get_va(TZASC_BASE, MEM_AREA_IO_SEC);
+
+	if (!addr[0])
+		panic("Cannot get TZASC1 base address");
+
+#ifdef TZASC2_BASE
 	addr[1] = core_mmu_get_va(TZASC2_BASE, MEM_AREA_IO_SEC);
 
-	for (i = 0; i < 2; i++) {
-		uint8_t region = 1;
+	if (!addr[1])
+		panic("Cannot get TZASC2 base address");
+
+	tzc_controllers = 2;
+#endif
+
+	for (i = 0; i < tzc_controllers; i++) {
+		region = 1;
 
 		tzc_init(addr[i]);
 
-		region = tzc_auto_configure(CFG_DRAM_BASE, CFG_DDR_SIZE,
-			     TZC_ATTR_SP_NS_RW, region);
-		region = tzc_auto_configure(CFG_TZDRAM_START, CFG_TZDRAM_SIZE,
-			     TZC_ATTR_SP_S_RW, region);
-		region = tzc_auto_configure(CFG_SHMEM_START, CFG_SHMEM_SIZE,
-			     TZC_ATTR_SP_ALL, region);
-		DMSG("Action register: %xl", tzc_get_action());
+		region = tzc_auto_configure(CFG_TZC_NSEC_START, CFG_DDR_SIZE,
+					    TZC_ATTR_SP_NS_RW, region);
+		region = tzc_auto_configure(CFG_TZC_SEC_START, CFG_TZDRAM_SIZE,
+					    TZC_ATTR_SP_S_RW, region);
+		region = tzc_auto_configure(CFG_TZC_SHMEM_START, CFG_SHMEM_SIZE,
```

Comment:
I don't see why introducing a new `CFG_TZC_SHMEM_START` to reflect the value of `CFG_SHMEM_START`.
I expect they are the same since relying on the same size value that is `CFG_SHMEM_SIZE`.
If you think iMX need this `CFG_TZC_SHMEM_START`, maybe name it `CFG_IMX_TZC_SHMEM_START`.

This comments applies also to `CFG_TZC_NSEC_START` and `CFG_TZC_SEC_START`.

Reply:
I introduced these values because tzasc region start addresses on imx8 have an offset (-DRAM base address). This offset is not present on imx6/7. Instead of doing `#idef CFG_IMX6 ... #endif` in the `tzc380.c` to handle this offset thing, I introduced `CFG_TZC_SHMEM_START` to have a more generic tzasc initialization.
I can rename it `CFG_IMX_TZC_*` indeed since it is specific to imx.